doc/design/draft-metadata-migration.md

```diff
@@ -0,0 +1,13 @@
+Premise: Draft metadata should always be in the latest schema version.
+
+There are a few places in the API where draft metadata are altered (POST, PUT) and served (GET). Note that the UI and 
+API use a shared schema version. All actions should thus assume that this is the only version available to the UI and
+server. Any other versions need to be migrated to this version for any display or editing.
+
```

Comment:
```suggestion

This proposal only applies to the `schemaVersion` in dandiset metadata, not asset metadata.

```

Reply:
good suggestion, I would have made it a bit more specific

```suggestion

This proposal only applies to the `schemaVersion` of dandiset metadata, not asset metadata.

```

since it is not just about the value but of the schema version that record would be expressed in.